src/drivers/gitlab.js

```diff
@@ -244,15 +244,38 @@ class Gitlab {
     body.append('title', title);
     body.append('description', description);
 
-    const { web_url: url } = await this.request({
+    const { web_url: url, iid } = await this.request({
       endpoint,
       method: 'POST',
       body
     });
 
+    if (autoMerge) {
+      await new Promise((resolve) => setTimeout(resolve, 20000));
+      await this.prAutoMerge({ mergeRequestId: iid });
+    }
+
     return url;
   }
 
+  /**
+   * @param {{ mergeRequestId: string }} param0
+   * @returns {Promise<void>}
+   */
+  async prAutoMerge({ mergeRequestId }) {
+    const projectPath = await this.projectPath();
+
+    const endpoint = `/projects/${projectPath}/merge_requests/${mergeRequestId}/merge`;
+    const body = new URLSearchParams();
+    body.append('merge_when_pipeline_succeeds', true);
```

Comment:
@DavidGOrtega I think you missed this line. This PR does wait for CI to pass.

Reply:
Im discussing that precisely 🙂 if GH supports that. You might need to setup the checks and for that you need a premium/pro account

Reply:
@casperdcl

Reply:
GH free tier allows checks (on public repos only) AFAIK.

Reply:
Ok, just to clarify. In GH would work if:
 - your repo is public
 - your repo is pro/team... if its private

Reply:
yes exactly. Public and/or paid.

Reply:
To my mind, the most handy behavior would be [create pull request, try automatic merge and, if it fails, try direct merge](https://github.com/iterative/discord-corpus/blob/49556627a09a85dd8604d9be5b644a601107cdaa/.github/workflows/crawl.yaml#L43). Automatic merge may not be possible in some cases, but is the most convenient option.

Reply:
Doesn't that defeat the purpose of auto-merge?

Reply:
Not completely: if it's available, uses it; if not, tries to merge it immediately; if not, users can merge it manually.

Reply:
Ah you mean "create pull request, try automatic merge and, if not enabled, fallback direct merge"

Reply:
Yup, see the link above

Reply:
resolving so can be merged

Reply:
Worth opening a follow–up issue?

Reply:
Yes did #907 :)

Reply:
![Fastpoke](https://user-images.githubusercontent.com/11387611/157549599-74dd69fa-08b2-4793-aee6-910c8d569540.png)